lib/ansible/modules/cloud/google/gce_facts.py

```diff
@@ -0,0 +1,359 @@
+from ansible.module_utils.gcp import get_valid_location
+
+
+def get_instance_info(inst):
+    """Retrieves instance information from an instance object and returns it
+    as a dictionary.
+
+    """
+    metadata = {}
+    if 'metadata' in inst.extra and 'items' in inst.extra['metadata']:
+        for md in inst.extra['metadata']['items']:
+            metadata[md['key']] = md['value']
+
+    try:
+        netname = inst.extra['networkInterfaces'][0]['network'].split('/')[-1]
+    except:
+        netname = None
+    try:
+        subnetname = inst.extra['networkInterfaces'][0]['subnetwork'].split('/')[-1]
+    except:
+        subnetname = None
+    if 'disks' in inst.extra:
+        disk_names = [disk_info['source'].split('/')[-1]
+                      for disk_info
+                      in sorted(inst.extra['disks'],
+                                key=lambda disk_info: disk_info['index'])]
+    else:
+        disk_names = []
+
+    if len(inst.public_ips) == 0:
+        public_ip = None
+    else:
+        public_ip = inst.public_ips[0]
+
+    return({
+        'image': inst.image is not None and inst.image.split('/')[-1] or None,
+        'disks': disk_names,
+        'machine_type': inst.size,
+        'metadata': metadata,
+        'name': inst.name,
+        'network': netname,
+        'subnetwork': subnetname,
+        'private_ip': inst.private_ips[0],
+        'public_ip': public_ip,
+        'status': ('status' in inst.extra) and inst.extra['status'] or None,
+        'tags': ('tags' in inst.extra) and inst.extra['tags'] or [],
+        'zone': ('zone' in inst.extra) and inst.extra['zone'].name or None,
+    })
+
+
+def instance_info(module, gce, instance_names, number, zone):
+    """Collects the information of a list of instances.
+
+    module: Ansible module object
+    gce: authenticated GCE connection object
+    instance_names: a list of instance names to collect facts
+    zone: GCEZone object where the instances reside
+
+    Returns a list of dictionaries, each dictionary contains information
+    about the instance.
+
+    """
+    changed = False
+    node_info = []
+
+    if isinstance(instance_names, str) and number:
+        node_names = ['%s-%03d' % (instance_names, i) for i in range(number)]
+    elif isinstance(instance_names, str) and not number:
+        node_names = [instance_names]
+    else:
+        node_names = instance_names
+
+    for name in node_names:
+        inst = None
+        try:
+            inst = gce.ex_get_node(name, zone)
+        except Exception as e:
+            module.fail_json(msg=unexpected_error_msg(e), changed=False)
+        else:
+            node_info.append(get_instance_info(inst))
+    return (changed, node_info)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            instance_names=dict(),
+            name=dict(aliases=['base_name']),
+            num_instances=dict(type='int'),
+            zone=dict(default='us-central1-a'),
```

Comment:
What do you think about making this required=True and removing the default? I think it would be best to make sure that users are fully aware of which zone they're using. Many users don't use the us-central1-a zone.

Reply:
@rambleraptor agreed, Will do the changes.